### PR TITLE
CLDC-3669 Extract bulk upload validations

### PR DESCRIPTION
**Pull request by kosiakkatrina**

### CODE REVIEW

File: config/locales/validations/sales/2023/bulk_upload.en.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    sales:
+      2023:
+        bulk_upload:
+          blank_file: "Template is blank - The template must be filled in for us to create the logs and check if data is correct."
+          over_max_column_count: "Too many columns, please ensure you have used the correct template."
+          no_headers: "Your file does not contain the required header rows. Add or check the header rows and upload your file again. [Read more about using the template headers](%{guidance_link})."
+          wrong_field_numbers_count: "Incorrect number of fields, please ensure you have used the correct template."
+          wrong_template: "Incorrect sale dates, please ensure you have used the correct template."
+          not_answered: "You must answer %{question}" # this is the same as not default not answered for single log submission
+          invalid_option: "Enter a valid value for %{question}" # this is the same as not default not answered for single log submission
```
Comment (Dinssa) [inline config/locales/validations/sales/2023/bulk_upload.en.yml]: Do you want to keep the comments? They're a little unclear, "not default not answered"?

Comment (kosiakkatrina) [reply config/locales/validations/sales/2023/bulk_upload.en.yml]: Ah, yes, the comments are a bit wrong, I'll remove them!

File: app/services/bulk_upload/processor.rb [hunk 2]
```diff
@@ -3,12 +3,12 @@ class BulkUpload::Processor
...
   BLANK_TEMPLATE_ERRORS = [
     I18n.t("activemodel.errors.models.bulk_upload/lettings/validator.attributes.base.blank_file"),
-    I18n.t("activemodel.errors.models.bulk_upload/sales/validator.attributes.base.blank_file"),
+    I18n.t("validations.sales.2024.bulk_upload.blank_file"),
   ].freeze
 
   WRONG_TEMPLATE_ERRORS = [
     *I18n.t("activemodel.errors.models.bulk_upload/lettings/validator.attributes.base", default: {}).values,
-    *I18n.t("activemodel.errors.models.bulk_upload/sales/validator.attributes.base", default: {}).values,
+    *I18n.t("validations.sales.2024", default: {}).values,
```
Comment (Dinssa) [inline app/services/bulk_upload/processor.rb]: Does this "validations.sales.2024" need to be "validations.sales.2024.bulk_upload"? Will this store all the translations not just the wrong template errors? Also, do we want to hardcode this for 2024?

Comment (Dinssa) [reply app/services/bulk_upload/processor.rb]: Would it make sense to put blank_file; over_max_column_count; no_headers; wrong_field_numbers_count: and wrong_template: under another key under bulk upload e.g. `validations.sales.2024.bulk_upload.upload_errors.blank_file`

That way all the translations grabbed by `WRONG_TEMPLATE_ERRORS` would just be those 5.

Comment (kosiakkatrina) [reply app/services/bulk_upload/processor.rb]: All very good suggestions, added now!

**Review by Dinssa — APPROVED** [review]

**Pull request merged**